### PR TITLE
Add authentication for notifications service

### PR DESCRIPTION
**Pull request by Glipchik**
Add swagger authentication and cross-service interaction

### CODE REVIEW

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs [hunk 1]
```diff
@@ -42,6 +52,31 @@ public async Task<TicketCityViewModel> Create([FromBody] CreateUpdateTicketCityV
...
+                    ClientSecret = AuthValues.TicketsClientSecret,
+                    Scope = AuthValues.NotificationsScopeName
+                }, cancellationToken);
+
+            var notificationsClient = _httpClientFactory.CreateClient();
+
+            notificationsClient.SetBearerToken(tokenResponse.AccessToken);
+
+            await notificationsClient.PostAsJsonAsync(_configuration["Urls:NotificationsEvent"], ticketInfo, cancellationToken);
+
```
Comment (moltenessence) [inline AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs:79]: move this logic out of controller

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs [hunk 2]
```diff
@@ -42,6 +52,31 @@ public async Task<TicketCityViewModel> Create([FromBody] CreateUpdateTicketCityV
...
 
+            var city = await _cityService.Get(model.CityId, cancellationToken);
+            var ticketInfo = new NewTicketInfoViewModel()
+            {
+                StayingStatus = model.StayingStatus,
+                CityName = city.Name,
+            };
+
+            var authClient = _httpClientFactory.CreateClient();
+            var discoveryDocument = await authClient.GetDiscoveryDocumentAsync("https://localhost:5001", cancellationToken);
```
Comment (moltenessence) [inline AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs:63]: move this logic out of controller

File: AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs [hunk 3]
```diff
@@ -42,6 +52,31 @@ public async Task<TicketCityViewModel> Create([FromBody] CreateUpdateTicketCityV
...
+
+            var tokenResponse = await authClient.RequestClientCredentialsTokenAsync(
+                new ClientCredentialsTokenRequest
```
Comment (moltenessence) [inline AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs:66]: move this logic out of controller

**Review by moltenessence — CHANGES_REQUESTED** [review]

File: AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs (hunk 3 above)
Comment (Glipchik) [reply AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs]: Fixed

File: AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs (hunk 2 above)
Comment (Glipchik) [reply AirlineTickets/AirlineTickets.API/Controllers/AirlineTicketCityController.cs]: Fixed

File: AirlineTickets/AirlineTickets.BLL/Interfaces/IHttpClientService.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿namespace AirlineTickets.BLL.Interfaces
+{
+    public interface IHttpClientService
+    {
+        public Task<HttpClient> GetAuthNotificationsClientAsync(CancellationToken cancellationToken);
```
Comment (nantonov) [inline AirlineTickets/AirlineTickets.BLL/Interfaces/IHttpClientService.cs:5]: access modifier in interface?

**Review by moltenessence — APPROVED** [review]

File: AirlineTickets/AirlineTickets.BLL/Interfaces/IHttpClientService.cs (hunk 4 above)
Comment (Glipchik) [reply AirlineTickets/AirlineTickets.BLL/Interfaces/IHttpClientService.cs]: Fixed

**Review by Genresal — APPROVED** [review]

**Pull request merged**